src/components/Sections/SimpleMode/RenderListHelpers.js

```diff
@@ -130,37 +149,59 @@ const unitDerivedFiles = (unit, type, keyFilter, mimeFilter) => {
     : [];
 };
 
-// REFACTOR THIS TO COMMON LIBRARY
-// SHOULD CONFIDER ORIGINAL LANGUAGE TOO!
-const bestFileByContentLanguages = (files, contentLanguages, originalLanguage) => files
-  .filter(file => contentLanguages.includes(file.language))
-  .sort((a, b) => {
-    const media = sortMediaFiles(a, b);
-    if (media !== 0) {
-      return media;
+const sortByMediaFunc = (contentLanguages, originalLanguage) => (a, b) => {
+  // sort by media type (i.e. files in all languages are sorted by media type first)
+  const media = sortMediaFiles(a, b);
+  if (media !== 0) {
+    return media;
+  }
+
+  // inside media type group sort by language (original language is preferable)
+  const lang = contentLanguages.indexOf(a.language) - contentLanguages.indexOf(b.language);
+  if (lang !== 0) {
+    if (a.language === originalLanguage) {
+      return -1;
     }
 
-    const lang = contentLanguages.indexOf(a.language) - contentLanguages.indexOf(b.language)
-    if (lang !== 0) {
-      if (a.language === originalLanguage) {
-        return -1;
-      }
+    if (b.language === originalLanguage) {
+      return 1;
+    }
 
-      if (b.language === originalLanguage) {
-        return 1;
-      }
+    return lang;
+  }
 
-      return lang;
-    }
+  return 0;
+};
 
-    return 0;
-  }).filter((file, index, files) => index === 0 || sortMediaFiles(files[index-1], files[index]) !== 0)
+// TODO: bbdev REFACTOR THIS TO COMMON LIBRARY
+// Order files by types and languages; remove some duplicates
```

Comment:
Is this comment still true?

Reply:
It looks like yes, but you're welcome to suggest better wording.